src/widgets/medications/medication-level-two.component.tsx

```diff
@@ -31,62 +32,165 @@ export default function MedicationLevelTwo(props: MedicationsOverviewProps) {
+                    return (
+                      <React.Fragment key={medication.uuid}>
+                        <tr>
+                          <td>
+                            <span
+                              style={{
+                                fontWeight: 500,
+                                color: "var(--omrs-color-ink-high-contrast)"
+                              }}
+                            >
+                              {medication.drug.name}
+                            </span>
+                            {" \u2014 "} {medication.doseUnits.display}&nbsp;
+                            {" \u2014 "}
+                            {medication.route.display} {" \u2014 "}
+                            DOSE &nbsp;
+                            <span
+                              style={{
+                                fontWeight: 500,
+                                color: "var(--omrs-color-ink-high-contrast)"
+                              }}
+                            >
+                              {getDosage(
+                                medication.drug.strength,
+                                medication.dose
+                              )}
+                              &nbsp;
+                              {medication.frequency.display}
+                            </span>
+                          </td>
+                          <td>{medication.action}</td>
+                          <td>
+                            {dayjs(medication.dateActivated).format(
+                              "DD-MMM-YYYY"
+                            )}
+                          </td>
+                        </tr>
+                      </React.Fragment>
+                    );
+                  })}
+            </tbody>
           </table>
+          <div className={styles.medicationFooter}>
+            <p
+              style={{ color: "var(--omrs-color-ink-medium-contrast)" }}
+              className={"omrs-type-body-large"}
+            >
+              No more medications available.
+            </p>
+          </div>
         </SummaryCard>
       </React.Fragment>
     );
   }
 
-  function displayPreviousMedication() {
-    <SummaryCard
-      name={t("Medications-previous", "Medications-previous")}
-      match={props.match}
-      styles={{ width: "90%" }}
-      addBtnUrl={`/patient/${patientUuid}/chart/add`}
-    >
-      <table className={styles.medicationsTable}>
-        <thead>
-          <tr>
-            <td>Name</td>
-            <td>
-              <div className={styles.centerItems}>Status</div>
-            </td>
-            <td>Start Date</td>
-          </tr>
-        </thead>
-        <tbody>{patientMedications && parseMedication()}</tbody>
-      </table>
-      <div className={styles.medicationFooter}>
-        <p
-          style={{ color: "var(--omrs-color-ink-medium-contrast)" }}
-          className={"omrs-type-body-large"}
+  function displayPastMedications() {
```

Comment:
L35 method has a lot of similarities `displayPastMedication`. Is there a reason we should explicitly have two methods for past and current medication?

Reply:
I understand your concern, I tried that at first but it makes the logic complicated, the functions are similar but they have major differences that would make it complicated for no good reason.Let me know your thoughts on this.